src/System.Xml.XmlSerializer/src/System/Xml/Serialization/CodeIdentifiers.cs

```diff
@@ -79,7 +78,7 @@ public string AddUnique(string identifier, object value)
         /// </devdoc>
         public bool IsInUse(string identifier)
         {
-            return _identifiers.Contains(identifier) || _reservedIdentifiers.Contains(identifier);
+            return _identifiers.ContainsKey(identifier) || _reservedIdentifiers.ContainsKey(identifier);
```

Comment:
`_reservedIdentifiers` is never written to (stays empty), and `_list` is never read from.
For that matter, there's no way to retrieve the stored object either - the whole function of this class is to get the next unique key - so you can just switch to a single set internally.